### PR TITLE
adding ADRs to wasmCloud repository

### PR DESCRIPTION
**Pull request by autodidaddict**
This pull request migrates the ADRs from the old ADR repository to the wasmCloud repository.
In addition, I've added the ADR that describes the decision to enforce the single-version-per-lattice entity rule.

### CODE REVIEW

File: adr/0007-tenancy.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+## Context and Problem Statement
+
+When running wasmCloud, people often need to run multiple copies of the same actor or capability provider. Running multiple copies of the capability provider with different link names gives us the ability to have multiple message brokers per actor, or multiple key-value stores, etc.
+
+When we run multiple copies of the same actor within the same host, we assume that these copies are _solely_ for horizontally scaling actors, and _not_ for multi-tenancy.
+
+An **inviolable** rule of wasmCloud's zero-trust policy is that we absolutely cannot ever allow one actor to compromise the security of another. We can't allow the generation of a link definition at runtime that could create a potential data exfiltration or PII compromise situation, or even enable the sharing of seemingly harmless data.
+
+Given the state of our documentation, people looking for multi-tenant solutions might not have sufficient context to make the right architectural decisions, and so this ADR needs to make clear where the tenancy boundaries are so that they can in turn be documented clearly at [wasmcloud.dev](https://wasmcloud.dev).
```
Comment (aschrijver) [inline adr/0007-tenancy.md:14]: wasmcloud.dev redirects to wasmcloud.com (might update URL if that's the new domain).

Comment (autodidaddict) [reply adr/0007-tenancy.md:14]: Good catch. This is copied text from an old adr from when the .dev site was different.

Comment (aschrijver) [reply adr/0007-tenancy.md:14]: Wonderful to have these ADR's.. invaluable insight into the decision making. Another :+1: for wasmCloud :)

Comment (autodidaddict) [reply adr/0007-tenancy.md]: Updated the links

**Review by connorsmith256 — APPROVED** [review]
One Repo to rule ADRs all, One Repo to find them,
One Repo to bring them all and in the open bind them
In the Land of GitHub where the wasmCloud lies.

**Comment by autodidaddict** [comment]
> One Repo to rule ADRs all, One Repo to find them, One Repo to bring them all and in the open bind them In the Land of GitHub where the wasmCloud lies.

Why is there no :sauron: emoji?? Not even a lord of the 💍 s

**Pull request merged**